### PR TITLE
Avoid some first-chance exceptions by returning instead

### PR DESCRIPTION
**Pull request by sharwell**
Observed a few hundred exceptions in a performance trace that were easy to clean up.

### CODE REVIEW

File: src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs [hunk 1]
```diff
@@ -100,7 +108,15 @@ private async ValueTask SelectItemAsync(CancellationToken cancellationToken)
         {
             // Switch to the UI so we can determine where the user is and determine the state the last time we updated
             // the UI.
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+#pragma warning disable VSTHRD004 // Await SwitchToMainThreadAsync
```
Comment (ToddGrun) [inline src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs:111]: Should we push to have the analyzer changed to recognize this pattern?

Here, right? [https://github.com/microsoft/vs-threading/blob/main/src/Microsoft.VisualStudio.Threading.Analyzers/AbstractVSTHRD004AwaitSwitchToMainThreadAsyncAnalyzer.cs](https://github.com/microsoft/vs-threading/blob/main/src/Microsoft.VisualStudio.Threading.Analyzers/AbstractVSTHRD004AwaitSwitchToMainThreadAsyncAnalyzer.cs)

File: src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs [hunk 2]
```diff
@@ -100,7 +108,15 @@ private async ValueTask SelectItemAsync(CancellationToken cancellationToken)
...
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken).NoThrowAwaitable();
```
Comment (ToddGrun) [inline src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs:112]: >SwitchToMainThreadAsync [](http://example.com/codeflow?start=56&length=23)

While this file is being modified, would it make sense to also change the SwitchToMainThreadAsync call in SelectItemWorkerAsync?

Comment (sharwell) [reply src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs:112]: These were limited to specific cases based on a performance trace. The other method either wasn't showing up in that trace, or didn't follow the same pattern of having an ignored return value.

File: src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs (hunk 1 above)
Comment (sharwell) [reply src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs:111]: Yes, I filed a report previously and the issue has already been fixed in code.

**Review by jasonmalinowski — APPROVED** [review]

**Review by ToddGrun — APPROVED** [review]

File: src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs (hunk 1 above)
Comment (ToddGrun) [reply src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs]: Looks great, thanks!

**Review by 333fred — COMMENTED** [review]
Is this new analyzer version analyzing local functions for `Async` suffixes, and that's why you're making name-only changes in the compiler layer?

**Comment by sharwell** [comment]
@333fred yes, that's exactly why the names changed.

**Review by jaredpar — APPROVED** [review]
Reviewed compiler item

**Review by 333fred — APPROVED** [review]
Compiler changes LGTM (though I must admit I dislike the Async suffix convention in general).

**Pull request merged**